environs/manual/linux/provisioner.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2016 Canonical Ltd.
+// Copyright 2016 Cloudbase Solutions SRL
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package linux
+
+import (
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/environs/manual/common"
+
+	"github.com/juju/loggo"
+)
+
+var (
+	logger = loggo.GetLogger("juju.environs.manual.linux")
+)
+
+const Scope = "ssh"
+
+type provision struct {
```

Comment:
I'm using this struct because this patch was made from another big patch, that I intent to soon release this main PR.
https://github.com/hoenirvili/juju/pull/8
with this patch also in utils.
https://github.com/hoenirvili/utils/commit/0a87d413aff364e0d0cf96bfeb90f5d894acbead


The patch enables **manual provisioning for windows machines**. In this moment, we can manually provision just ```linux like``` machines. @kat-coif you look closely in the manual/provisioner.go we have this interface to let us more easily add more implementations for this process of provisioning.

```go
// Provisioner is the process of provisioning a windows/linux machine                                                                                                                                                
type Provisioner interface {                                                                                                                                                                                         
    // ProvisionMachine is the main entrypoint of the Provisioner                                                                                                                                                                                                                                                                                              
    ProvisionMachine(params common.ProvisionMachineArgs) (machineId string, err error)                                                                                                                               
}     
```